.github/workflows/gpu-smoke-test.yml

```diff
@@ -0,0 +1,113 @@
+      with:
+        ref: ${{ github.ref }}
+    - uses: actions/setup-go@v3
+      with:
+        go-version-file: go.mod
+    - if: matrix.provider == 'AWS'
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-region: us-west-1
+        role-to-assume: arn:aws:iam::342840881361:role/SandboxUser
+    - run: go test ./task -v -timeout=30m -count=1 -tags=smoke
+    - uses: actions/checkout@v3
+      with:
+        ref: master
+    - run: go test ./task -v -timeout=30m -count=1 -tags=smoke
+      env:
+        SMOKE_TEST_SWEEP: true
+        SMOKE_TEST_TARGET: gpu
+  test-k8s:
+    name: test (K8S)
+    needs: authorize
+    environment: automatic
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    concurrency: ${{ github.workflow }}-K8S
+    env:
+      AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+      AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+      AZURE_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha || github.ref }}
+    - uses: actions/setup-go@v3
+      with:
+        go-version-file: go.mod
+    - run: |
+        az login \
+          --service-principal \
+          --user="$AZURE_CLIENT_ID" \
+          --password="$AZURE_CLIENT_SECRET" \
+          --tenant="$AZURE_TENANT_ID"
+        az account set \
+          --subscription="$AZURE_SUBSCRIPTION_ID"
+    - id: cluster
+      run: |
+        az extension add \
+          --name=aks-preview
+        az provider register \
+          --namespace=Microsoft.ContainerService
+        az feature register \
+          --namespace=Microsoft.ContainerService \
+          --name=GPUDedicatedVHDPreview
+        az group create \
+          --name="tpiSmokeTestCluster$GITHUB_RUN_ID" \
+          --location=eastus
+        az aks create \
+          --resource-group="tpiSmokeTestCluster$GITHUB_RUN_ID" \
+          --name="tpiSmokeTestCluster$GITHUB_RUN_ID" \
+          --node-vm-size=Standard_NC6 \
+          --node-count=1 \
+          --aks-custom-headers=UseGPUDedicatedVHD=true \
+          --generate-ssh-keys
+        az aks get-credentials \
+          --resource-group="tpiSmokeTestCluster$GITHUB_RUN_ID" \
+          --name="tpiSmokeTestCluster$GITHUB_RUN_ID" \
+          --file - |
+        perl -0777p \
+          -e 's/%/%25/g;' \
+          -e 's/\n/%0A/g;' \
+          -e 's/\r/%0D/g;' \
+          -e 's/(.+)/::add-mask::\1\n::set-output name=kubeconfig::\1\n/g'
+    - run: go test ./task -v -timeout=30m -count=1 -tags=smoke
+      env:
+        KUBECONFIG_DATA: ${{ steps.cluster.outputs.kubeconfig }}
+        SMOKE_TEST_ENABLE_K8S: true
+        SMOKE_TEST_TARGET: gpu
+    - if: always()
+      run: |
+        az group delete \
+          --name="tpiSmokeTestCluster$GITHUB_RUN_ID" \
+          --yes
+env:
+  SMOKE_TEST_IDENTIFIER: smoke test ${{ github.run_id }}
```

Comment:
In continuous integration workflows, a little dependency is much worse than a ~little~ huge copying. Consider using [`if`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif) or other gating mechanisms to reuse the existing workflow.

Reply:
updated